File: Python/Q31.py
# Q31.py - Program to Make a Simple Calculator Using switch...case (Python 3.10+)
def calculator(a, b, op):
    match op:
        case '+': return a + b
        case '-': return a - b
        case '*': return a * b
        case '/': return a / b if b != 0 else 'Division by zero'
        case '%': return a % b
        case _: return 'Invalid operation'

a = float(input("Enter first number: "))
b = float(input("Enter second number: "))
op = input("Enter operation (+, -, *, /, %): ")
print(f"Result: {calculator(a, b, op)}")

File: C/Q31.c
#include <stdio.h>

int main() {
    char operator;
    double num1, num2;

    scanf(" %c", &operator); 
    scanf("%lf %lf", &num1, &num2); 

    switch (operator) {
        case '+':
            printf("Result: %.2lf", num1 + num2);
            break;
        case '-':
            printf("Result: %.2lf", num1 - num2);
            break;
        case '*':
            printf("Result: %.2lf", num1 * num2);
            break;
        case '/':
            if (num2 != 0)
                printf("Result: %.2lf", num1 / num2);
            else
                printf("Error: Division by zero");
            break;
        default:
            printf("Invalid operator");
    }

    return 0;
}
